components/pages/searchPageLayout/SearchPageLayout.tsx

```diff
@@ -70,43 +67,56 @@ const SearchPageLayout = ({ searchQuery }: { searchQuery?: string }) => {
           [mapFacetsToFilters[facetDefinition]]: [...values],
         }
       }
-
       return acc
     },
     {} as { [key: string]: keyof SearchFilters[] }
   )
 
-  const { data, isLoading } = useSearchWithPaginationQuery({
-    q: { all: q },
-    offset: 0,
-    limit: 10,
-    filters: {
-      branchId: branchIds,
-      ...facetsForSearchRequest,
+  const { data, isLoading } = useSearchWithPaginationQuery(
+    {
+      q: { all: q },
+      offset: 0,
+      // TODO: This should be configurable.
+      limit: 10,
+      filters: {
+        branchId: branchIds,
+        ...facetsForSearchRequest,
+      },
     },
-  })
+    {
+      enabled: q?.length > 0,
```

Comment:
Should be greater than 2 right?
And here we are back to some static configuration :)

Reply:
I'd say for now we can keep it more than 0. I'd say because the target audience is children they might not be able to easily understand that they need min 3 letters. And we can totally make queries with just 1 letter.